src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -2228,6 +2228,26 @@ public static Map<String,String> getInvertiblePinMap(Series series, Unisim unisi
 		return invertPinMap;
 	}
 
+	/**
+	 * Copy a cell from a design as a new design.
+	 * @param src The source design with or without implementation
+	 * @param cellName The full hierarchy cell name to be extracted as another design
+	 * @param  keepStaticRouting A flag to indicate whether to keep the routes of static nets or not.
+	 * @return A newly created design copied from the source
+	 */
+	public static Design createDesignFromCell(Design src, String cellName, boolean keepStaticRouting) {
+		EDIFNetlist srcCellNetlist = EDIFTools.createNewNetlist(src.getNetlist().getHierCellInstFromName(cellName).getInst());
+		EDIFTools.ensureCorrectPartInEDIF(srcCellNetlist, src.getPartName());
+		Design d2 = new Design(srcCellNetlist);
+		d2.setAutoIOBuffers(false);
+		d2.setDesignOutOfContext(true);
+
+		// TODO: Skip this step if the design was not implemented.
```

Comment:
This check seems fairly straight forward.  Can we add it now?

Reply:
@clavin-xlnx please add it.  I thought it is possible, but don't know how.

Reply:
TODO: Check if there any `SiteInst`-s there are in the `Design`. If there are none, then design is unplaced.